### DISCUSSION TITLE
Suggestions to update the contribution guide

### DISCUSSION THREAD

**Discussion by azzenabidi** [erikhuda/thor#840] [Ideas]
I have been navigating Thor's codebase as I am working on #666. I am still trying  to fully understand how the pieces work together before building the feature. 

The current [contribution document](https://github.com/rails/thor/blob/main/CONTRIBUTING.md) doesn't help much. It would be helpful to have an extensive guide that covers the following:
- Dev environment setup
- Coding Conventions 
- Running tests
- A high level overview of the codebase(entry point + common scenarios to consider when adding a new feature etc.).
- How to contribute to documentation ( report bugs properly, update the wiki and the website content etc.).

I love Thor and I am definitely interested in working on the contribution guide as it will help tackle issues faster and welcome new contributors to the project. What are your thoughts? Thanks!

**Comment by dorner** [comment]
We're happy for any contributions! Give it a whirl and we'll review!

**Comment by patrick-motard** [comment]
@azzenabidi I am interested in trying out whatever you end up writing up. I've submitted [an issue](https://github.com/rails/thor/issues/768) that I would like to fix. I also would like to nudge along the many open PRs on this project. From the sounds of it, [this project could use some love](https://github.com/rails/thor/issues/491). Any efforts to ease the onboarding process would go a long way.

**Comment by azzenabidi** [comment]
@patrick-motard Glad to hear that! I am making progress with the guide. If everything goes well, I will be opening a pull request next week :) cc @dorner

**Comment by dorner** [comment]
Nice, looking forward!
